src/components/SearchBar.jsx

```diff
@@ -0,0 +1,72 @@
+// LIBRARY IMPORTS
+import React, { useState, useRef } from 'react';
+import { MagnifyingGlassIcon, XMarkIcon } from '@heroicons/react/24/solid';
+
+// LOCAL IMPORTS
+
+export default function SearchBar({ data, setData, setClearedSearch, fullDataSet }) {
+  // SET STATES
+  const [searchSubmitted, setSearchSubmitted] = useState(false);
+  const [searchTerm, setSearchTerm] = useState('');
+  const [searchSuccessful, setSearchSuccessful] = useState(false);
+  const searchTermRef = useRef('');
+
+  // EVENT HANDLERS
+  const handleInputChange = (event) => {
+    setSearchTerm(event.target.value);
+    searchTermRef.current = event.target.value;
+  };
+  const handleInputSubmit = (event) => {
+    event.preventDefault();
+    const results = data.filter((item) =>
+      item.name.toLowerCase().includes(searchTerm.toLowerCase().trim())
+    );
+    setData(results);
+    if (results.length === 0) {
+      setClearedSearch(false);
+    }
+    if (results.length > 0) {
+      setSearchSuccessful(true);
+    }
+    setSearchSubmitted(true);
+    setSearchTerm('');
+  }
+  const handleClearSubmit = (event) => {
+    event.preventDefault();
+    setData(fullDataSet);
+    setSearchTerm('');
+    setClearedSearch(true);
+    setSearchSubmitted(false);
+    setSearchSuccessful(false);
+  }
+
+
+  return (
+    <>
+      <form className=" w-full m-auto max-w-lg mt-2 relative" onSubmit={handleInputSubmit}>
```

Comment:
conditionally setting `onSubmit` to two different functions was causing a lot of bugs, so instead I set the default to one function and then handled it conditionally changing to the clear function below.